include/xlnt/utils/optional.hpp

```diff
@@ -34,110 +35,282 @@ namespace xlnt {
 /// within the optional class.
 /// </summary>
 template <typename T>
-class XLNT_API optional
+class optional
 {
+#if _MSC_VER <= 1900 // v14, visual studio 2015
+#define XLNT_NOEXCEPT_VALUE_COMPAT(...) (false)
```

Comment:
I can't say I like this macro, but noexcept on vs2015 (CI) just didnt want to play ball. All noexcepts are appropriately conditional on clang/GCC and vs2017